src/pages/activity.js

```diff
@@ -336,59 +283,54 @@ function TokenRow({ transfer, dateFormat }) {
 }
 
 function TokenRowDescription({ transfer }) {
-  const type = transferType(transfer);
   return (
     <div className="description">
-      {type === 'Migration' ? (
+      {transfer.type === 'MIGRATION' ? (
         <span>
           POAP migrated to
           <a
-            href={`${POAP_APP_URL}/scan/${transfer.to.id}`}
+            href={`${POAP_APP_URL}/scan/${transfer.to}`}
             target="_blank"
             rel="noopener noreferrer"
           >
             {' '}
-            {transfer.to.id.substring(0, 16) + '…'}{' '}
+            {transfer.to.substring(0, 16) + '…'}{' '}
           </a>
-          from {transfer.network} to Ethereum
+          from {transfer.chain} to Ethereum
         </span>
-      ) : type === 'Claim' ? (
+      ) : transfer.type === 'CLAIM' ? (
         <span>
           POAP claimed on event{' '}
-          <Link to={`/event/${transfer.token.event.id}`}>
-            #{transfer.token.event.id}
-          </Link>{' '}
-          on {transfer.network}
+          <Link to={`/event/${transfer.eventId}`}>#{transfer.eventId}</Link> on{' '}
+          {transfer.chain}
         </span>
-      ) : type === 'Burn' ? (
+      ) : transfer.type === 'BURN' ? (
         <span>
           POAP burned on event{' '}
-          <Link to={`/event/${transfer.token.event.id}`}>
-            #{transfer.token.event.id}
-          </Link>{' '}
-          on {transfer.network}
+          <Link to={`/event/${transfer.eventId}`}>#{transfer.eventId}</Link> on{' '}
+          {transfer.chain}
         </span>
       ) : (
         <span>
           POAP transferred from
           <a
-            href={`${POAP_APP_URL}/scan/${transfer.from.id}`}
+            href={`${POAP_APP_URL}/scan/${transfer.from}`}
             target="_blank"
             rel="noopener noreferrer"
           >
             {' '}
-            {shrinkAddress(transfer.from.id, 10)}{' '}
+            {shrinkAddress(transfer.from, 10)}{' '}
           </a>{' '}
           to
           <a
-            href={`${POAP_APP_URL}/scan/${transfer.to.id}`}
+            href={`${POAP_APP_URL}/scan/${transfer.to}`}
             target="_blank"
             rel="noopener noreferrer"
           >
             {' '}
-            {shrinkAddress(transfer.to.id, 10)}
+            {shrinkAddress(transfer.to, 10)}
```

Comment:
Why? Is because you don't want to write `transfer.` anymore? Hint: is already written. Or is because you comprehend better the code `{shrinkAddress(to, 10)}` than `{shrinkAddress(transfer.to, 10)}`? What about ```<Pill
              className={type}
              text={type}
              tooltip={false}
            />``` vs ```<Pill
              className={transfer.type}
              text={transfer.type}
              tooltip={false}
            />```?

I am not against if you know the **why**. Anyway the `transfer` variable need to be without destructing because is passed to `<TokenRowDescription transfer={transfer} />`, we can also deconstruct them if it helps you. I don't mind adding more memory footprint to people's browsers, this will not be run in a server so I don't care if it is bloated.

@rlajous

Reply:
Dismiss it